artifactory/commands/npm/installorci.go

```diff
@@ -506,59 +529,84 @@ func (nca *NpmCommandArgs) parseDependencies(data []byte, scope string) error {
-	dependencyKey := string(key) + "-" + string(ver)
+	dependencyKey := string(key) + ":" + string(ver)
 	if nca.dependencies[dependencyKey] == nil {
 		nca.dependencies[dependencyKey] = &dependency{name: string(key), version: string(ver), scopes: []string{scope}}
 	} else if !scopeAlreadyExists(scope, nca.dependencies[dependencyKey].scopes) {
 		nca.dependencies[dependencyKey].scopes = append(nca.dependencies[dependencyKey].scopes, scope)
 	}
 }
 
-// Creates a function that fetches dependency data from Artifactory. Can be applied from a producer-consumer mechanism
-func (nca *NpmCommandArgs) createGetDependencyInfoFunc(servicesManager artifactory.ArtifactoryServicesManager) getDependencyInfoFunc {
+// Creates a function that fetches dependency data.
+// If a dependency was included in the previous build, take the checksums information from it.
+// Otherwise, fetch the checksum from Artifactory.
+// Can be applied from a producer-consumer mechanism.
+func (nca *NpmCommandArgs) createGetDependencyInfoFunc(servicesManager artifactory.ArtifactoryServicesManager,
+	previousBuildDependencies map[string]*buildinfo.Dependency) getDependencyInfoFunc {
 	return func(dependencyIndex string) parallel.TaskFunc {
 		return func(threadId int) error {
 			name := nca.dependencies[dependencyIndex].name
 			ver := nca.dependencies[dependencyIndex].version
-			log.Debug(clientutils.GetLogMsgPrefix(threadId, false), "Fetching checksums for", name, "-", ver)
-			stream, err := servicesManager.Aql(serviceutils.CreateAqlQueryForNpm(name, ver))
-			if err != nil {
-				return err
-			}
-			defer stream.Close()
-			result, err := ioutil.ReadAll(stream)
-			if err != nil {
+
+			// Get dependency info.
+			checksum, fileType, err := getDependencyInfo(name, ver, previousBuildDependencies, servicesManager, threadId)
+			if err != nil || checksum == nil {
 				return err
 			}
-			parsedResult := new(aqlResult)
-			if err = json.Unmarshal(result, parsedResult); err != nil {
-				return errorutils.CheckError(err)
-			}
-			if len(parsedResult.Results) == 0 {
-				log.Debug(clientutils.GetLogMsgPrefix(threadId, false), name, "-", ver, "could not be found in Artifactory.")
-				return nil
-			}
-			nca.dependencies[dependencyIndex].artifactName = parsedResult.Results[0].Name
-			nca.dependencies[dependencyIndex].checksum =
-				&buildinfo.Checksum{Sha1: parsedResult.Results[0].Actual_sha1, Md5: parsedResult.Results[0].Actual_md5}
-			log.Debug(clientutils.GetLogMsgPrefix(threadId, false), "Found", parsedResult.Results[0].Name,
-				"sha1:", parsedResult.Results[0].Actual_sha1,
-				"md5", parsedResult.Results[0].Actual_md5)
+
+			// Update dependency.
+			nca.dependencies[dependencyIndex].fileType = fileType
+			nca.dependencies[dependencyIndex].checksum = checksum
 			return nil
 		}
 	}
 }
 
+// Get dependency's checksum and type.
+func getDependencyInfo(name, ver string, previousBuildDependencies map[string]*buildinfo.Dependency,
```

Comment:
Since this function is declared with named return values, there's no need to add them to the return statements.

Reply:
Changed the return statements.